Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
+	void FrameMan::Draw() {
+		// Count how many split screens we'll need
+		int screenCount = (m_HSplit ? 2 : 1) * (m_VSplit ? 2 : 1);
 
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
+		RTEAssert(screenCount <= 1 || m_pPlayerScreen, "Splitscreen surface not ready when needed!");
 
-		circle(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
-		circle(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
-	}
-}
+		// Clear out the post processing screen effects list
+		g_PostProcessMan.Reset();
 
+		// These accumulate the effects for each player's screen area, and are then transferred to the above lists with the player screen offset applied
+		list<PostEffect> screenRelativeEffects;
+		list<Box> screenRelativeGlowBoxes;
 
-void FrameMan::CircleFillPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) 
-{
-	if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) 
-	{ 
-		Vector drawStart = m_Start - targetPos; 
-		 
-		circlefill(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
-	} else {
-		Vector drawStartLeft;
-		Vector drawStartRight;
+		// Handy handle
+		Activity *pActivity = g_ActivityMan.GetActivity();
 
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
+		for (int whichScreen = 0; whichScreen < screenCount; ++whichScreen) {
+			// Screen Update
+			screenRelativeEffects.clear();
+			screenRelativeGlowBoxes.clear();
 
-		circlefill(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
-		circlefill(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
-	}
-}
-
-void FrameMan::TextPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) 
-{
-	if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) 
-	{ 
-		Vector drawStart = m_Start - targetPos; 
-		AllegroBitmap pPlayerGUIBitmap(pDrawScreen);
-
-		if (m_IsSmall)
-			g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStart.m_X, drawStart.m_Y, m_Text, m_Alignment);
-		else
-			g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStart.m_X, drawStart.m_Y, m_Text, m_Alignment);
-	} else {
-		Vector drawStartLeft;
-		Vector drawStartRight;
-
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
-
-		AllegroBitmap pPlayerGUIBitmap(pDrawScreen);
-		if (m_IsSmall)
-		{
-			g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStartLeft.m_X, drawStartLeft.m_Y, m_Text, m_Alignment);
-			g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStartRight.m_X, drawStartRight.m_Y, m_Text, m_Alignment);
-		}
-		else
-		{
-			g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStartLeft.m_X, drawStartLeft.m_Y, m_Text, m_Alignment);
-			g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStartRight.m_X, drawStartRight.m_Y, m_Text, m_Alignment);
-		}
-	}
-}
+			BITMAP *pDrawScreen = /*get_color_depth() == 8 && */screenCount == 1 ? m_pBackBuffer8 : m_pPlayerScreen;
```

Comment:
Kill this commented code. It looks like it's a holdover from before 32bpp bitmaps were supported or something. Either way, it's not useful and I very much doubt it ever will be.